builder/dockerfile/parser/line_parsers.go

```diff
@@ -168,6 +168,10 @@ func parseNameVal(rest string, key string, d *Directive) (*Node, map[string]bool
 			}
 			parts := strings.SplitN(word, "=", 2)
 
+			if len(parts[0]) == 0 {
+				return nil, nil, fmt.Errorf("Syntax error - ENV names can't be blank")
```

Comment:
`ENV` shouldn't be hardcoded here though (it's used for other directive). Probably should be `fmt.Errorf("Syntax error - %s names can't be blank", key)`